shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -157,53 +159,64 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     int replication = FileSystemShellUtils.getIntArg(cl, REPLICATION_OPTION, DEFAULT_REPLICATION);
     Set<String> workerSet = new HashSet<>();
+    Set<String> excludeWorkerSet = new HashSet<>();
     Set<String> localityIds = new HashSet<>();
     if (cl.hasOption(HOST_FILE_OPTION.getLongOpt())) {
       String hostFile = cl.getOptionValue(HOST_FILE_OPTION.getLongOpt()).trim();
-      readLinesToSet(workerSet, hostFile);
+      readLinesToSet(workerSet, excludeWorkerSet, hostFile);
     } else if (cl.hasOption(HOSTS_OPTION.getLongOpt())) {
       String argOption = cl.getOptionValue(HOSTS_OPTION.getLongOpt()).trim();
-      readItemsFromOptionString(workerSet, argOption);
+      readItemsFromOptionString(workerSet, excludeWorkerSet, argOption);
     }
     if (cl.hasOption(LOCALITY_FILE_OPTION.getLongOpt())) {
       String localityFile = cl.getOptionValue(LOCALITY_FILE_OPTION.getLongOpt()).trim();
-      readLinesToSet(localityIds, localityFile);
+      readLinesToSet(localityIds, null, localityFile);
     } else if (cl.hasOption(LOCALITY_OPTION.getLongOpt())) {
       String argOption = cl.getOptionValue(LOCALITY_OPTION.getLongOpt()).trim();
-      readItemsFromOptionString(localityIds, argOption);
+      readItemsFromOptionString(localityIds, null, argOption);
     }
 
     if (!cl.hasOption(INDEX_FILE.getLongOpt())) {
       AlluxioURI path = new AlluxioURI(args[0]);
-      DistributedLoadUtils.distributedLoad(this, path, replication, workerSet, localityIds);
+      DistributedLoadUtils.distributedLoad(this, path, replication, workerSet,
+          excludeWorkerSet, localityIds);
     } else {
       try (BufferedReader reader = new BufferedReader(new FileReader(args[0]))) {
         for (String filename; (filename = reader.readLine()) != null; ) {
           AlluxioURI path = new AlluxioURI(filename);
-          DistributedLoadUtils.distributedLoad(this, path, replication, workerSet, localityIds);
+          DistributedLoadUtils.distributedLoad(this, path, replication, workerSet,
+              excludeWorkerSet, localityIds);
         }
       }
     }
     return 0;
   }
 
-  private void readItemsFromOptionString(Set<String> localityIds,
+  private void readItemsFromOptionString(Set<String> localityIds, Set<String> excludedSet,
       String argOption) {
     for (String locality : StringUtils.split(argOption, ",")) {
       locality = locality.trim().toUpperCase();
       if (!locality.isEmpty()) {
-        localityIds.add(locality);
+        if (locality.charAt(0) == '!' && excludedSet != null) {
+          excludedSet.add(locality);
```

Comment:
if the `locality` starts with `!` , than you added into the excludeSet is started with `!`, so I guess you want to remove the magic char before put it into the excludedSet?